content/ember-4-5-released.md

```diff
@@ -0,0 +1,151 @@
+---
+title: Ember 4.5 Released
+authors:
+  - jen-weber
+  - chris-krycho
+date: 2022-07-12T00:00:00.000Z
+tags:
+  - releases
+  - '2022'
+  - version-4-x
+---
+
+Today the Ember project is releasing version 4.5 of Ember.js, Ember Data, and Ember CLI.
+
+This release kicks off the 4.5 beta cycle for all sub-projects. We encourage our community (especially addon authors) to help test these beta builds and report any bugs before they are published as a final release in six weeks' time. The [ember-try](https://github.com/ember-cli/ember-try) addon is a great way to continuously test your projects against the latest Ember releases.
+
+You can read more about our general release process here:
+
+- [Release Dashboard](http://emberjs.com/releases/)
+- [The Ember Release Cycle](https://blog.emberjs.com/new-ember-release-process/)
+- [The Ember Project](https://blog.emberjs.com/ember-project-at-2-0/)
+- [Ember LTS Releases](https://blog.emberjs.com/announcing-embers-first-lts/)
+
+---
+
+## Ember.js
+
+Ember.js is the core framework for building ambitious web applications.
+
+### Changes in Ember.js 4.5
+
+Ember.js 4.5 is an incremental, backwards compatible release of Ember with bug fixes, performance improvements, and minor deprecations.
+
+#### Bug Fixes
+
+Ember.js 4.5 introduced 0 bug fixes.
+
+#### Features
+
+Ember.js 4.5 introduced 2 new features.
+
+1. Plain function as helpers
+2. A new `renderSettled` test helper
+
+##### 1. Plain functions as helpers
+
+You can now use plain functions as helpers in your component templates. This helps make the relationship between Ember component templates and their JavaScript class more intuitive.
+
+For example, here we create a method `double` and use it directly in a template:
+
+```js
+// my-component.js
+
+import Component from '@glimmer/component';
+
+export default class MyComponent extends Component {
+  double = num => num * 2;
+}
+```
+
+```hbs
+// my-component.hbs
+
+{{this.double 2}}
+
+<SomeComponent @foo={{this.double 2}} />
+```
+
+Previously, you would need to [write a separate helper](https://guides.emberjs.com/release/components/helper-functions/#toc_writing-a-helper-function) in order to accomplish this.
```

Comment:
This isn't totally true. You could do the following, as of 3.25:

```js
export default class MyComponent extends Component {
  double = helper(([num]) => num * 2);
}
```

Reply:
Some common feedback I've gotten from co-workers, is that they're blown away by the DX of "anything as values from anywhere" and how it's a massive quality of life improvement.

They ask for guides / docs on this behavior, and I'm just like :shrug: idk what to tell them, because idk... _you *just* assign the value_.

Maybe since the guides are somewhat focused on globally available everything, they think local stuff isn't possible? idk.

Maybe there is a way to address? idk if this blog post is the right place to mention all this, but. yeah. lots of idk :upside_down_face:

Reply:
Thank you, I’ll incorporate this!

Reply:
I dropped “separate” and kept the link to the guides as-is. Does that seem ok?

Reply:
I think my gut recation is to the word "need"?

maybe 

>  Even though local helpers have been supported since 3.25, the documentation says to [write a separate helper](https://guides.emberjs.com/release/components/helper-functions/#toc_writing-a-helper-function) in order to accomplish this. 

maybe?

Reply:
I wonder if we inadvertently over-corrected for people being confused by data passing. I think we can address this confusion in the component guides, potentially.

Reply:
yeah potentially -- related, I ran in to someone who thought accessing args during initialization was impossible and sent them [this](https://limber.glimdown.com/?t=%23%20Limber%20Editor%0A%0AAccessing%20%60%40arg%60s%20during%20initialization.%0A%0A%60%60%60gjs%20live%20preview%0Aimport%20Component%20from%20%27%40glimmer%2Fcomponent%27%3B%0Aimport%20%7B%20tracked%20%7D%20from%20%27%40glimmer%2Ftracking%27%3B%0Aimport%20%7B%20on%20%7D%20from%20%27%40ember%2Fmodifier%27%3B%0A%0Aclass%20Child%20extends%20Component%20%7B%0A%20%20%2F%2F%20will%20not%20receive%20arg%20updates%20%0A%20%20%2F%2F%20(good%20for%20letting%20the%20user%20override%20passed%20in%20state)%0A%20%20%40tracked%20dedupedValue%3B%0A%0A%20%20constructor(owner%2C%20args)%20%7B%0A%20%20%20%20super(owner%2C%20args)%3B%0A%0A%20%20%20%20%2F%2F%20setup%20here%0A%20%20%20%20this.dedupedValue%20%3D%20args.value%3B%0A%20%20%7D%0A%0A%20%20update%20%3D%20(event)%20%3D%3E%20this.dedupedValue%20%3D%20event.target.value%3B%0A%0A%20%20%3Ctemplate%3E%0A%20%20%20%20%7B%7Bthis.dedupedValue%7D%7D%0A%0A%20%20%20%20%3Cbr%3E%0A%20%20%20%20%3Clabel%3E%0A%20%20%20%20%20%20Local%20change%0A%20%20%20%20%20%20%3Cinput%20%7B%7Bon%20%27input%27%20this.update%7D%7D%20class%3D%22border%22%3E%0A%20%20%20%20%3C%2Flabel%3E%0A%20%20%3C%2Ftemplate%3E%0A%7D%0A%0A%0A%3Ctemplate%3E%0A%20%20%3CChild%20%40value%3D%22hello%20there%22%20%2F%3E%0A%3C%2Ftemplate%3E%0A%60%60%60%0A)


It seems some folks somehow think data passing special and works different from what they expect.

Thankfully, since 3.25, all data/value passing works the same as it does in JS.

Reply:
I applied your wording. Thanks!